modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java

```diff
@@ -1,141 +1,353 @@
 	 *
-	 * @param clazz Class of the message object
-	 * @param msg   message
-	 * @param <T>   Generic type for the {@link Future}
-	 * @return a future which resolves when the decoding was successful
+	 * @return a future that represents the conversion state
 	 */
-	static <T extends JsonMessage> Future<T> on(Class<T> clazz, Message<?> msg) {
-		return on(clazz, msg.body());
+	static <T extends JsonMessage> Future<T> on(Class<T> type, String json) {
+		return Future.future(promise -> on(type, json, promise::complete, promise::fail));
 	}
 
 	/**
-	 * Creates a message from the given json representation.
+	 * Like {@link #on(Class, Object, Handler, Handler)} but returns a {@link Future} which resolves
+	 * when the conversion completes successfully or fails when a {@link DecodeException} or an
+	 * {@link IllegalArgumentException} occurs during conversion.
 	 *
-	 * @param json source
-	 * @param type class of the message
-	 * @param <T>  type of the {@link Message}
-	 * @return decoded message object
+	 * @return a future that represents the conversion state
 	 */
-	static <T extends JsonMessage> T from(String json, Class<T> type) {
-		return JSON_CODEC.fromString(json, type);
+	static <T extends JsonMessage> Future<T> on(Class<T> type, Object json) {
+		return Future.future(promise -> on(type, json, promise::complete, promise::fail));
 	}
 
 	/**
-	 * Creates a message from the given json representation.
+	 * Like {@link #on(Class, Message, Handler, Handler)} but returns a {@link Future} which resolves
+	 * when the conversion completes successfully or fails when a {@link DecodeException} occurs during conversion.
 	 *
-	 * @param json source which must be a String
-	 * @param type class of the message
-	 * @param <T>  type of the {@link Message}
-	 * @return decoded message object
+	 * @return a future that represents the conversion state
 	 */
-	static <T extends JsonMessage> T from(Object json, Class<T> type) {
-		return JSON_CODEC.fromString((String) json, type);
+	static <T extends JsonMessage> Future<T> on(Class<T> type, Message<?> message) {
+		return Future.future(promise -> on(type, message, promise::complete, promise::fail));
 	}
 
+	///////////////////////////////////////////////////////////////////////////
+	// synchronous decoding section
+	///////////////////////////////////////////////////////////////////////////
+
 	/**
-	 * Returns the simple class name of the subclass.
+	 * Constructs a {@link JsonMessage} from a buffer which contains an encoded JSON string.
 	 *
-	 * @return simple class name of subclass
+	 * @param json the buffer that contents contain the necessary information to construct
+	 *             the specified {@link JsonMessage}
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the buffer contents does not contain the necessary information to successfully
```

Comment:
```suggestion
	 * @throws DecodeException if the buffer contents do not contain the necessary information to successfully
```

either "buffer contents" and "do", or "buffer content" and "does" :D